docs/book/content/types/objects/using_contexts.md

```diff
@@ -87,4 +86,55 @@ impl User {
 You only get an immutable reference to the context, so if you want to affect
 change to the execution, you'll need to use [interior
 mutability](https://doc.rust-lang.org/book/first-edition/mutability.html#interior-vs-exterior-mutability)
-using e.g. `RwLock` or `RefCell`.
+using e.g. `RwLock` or `RefCell`. \
+If you are using async runtime like `tokio` for mutable references you will need to use a corresponding async version of RwLock:
+```rust
+# extern crate juniper;
+# use std::collections::HashMap;
+# use juniper::graphql_object;
+# use tokio::sync::RwLock;
+#
+// This struct represents our context.
+struct Database {
+    requested_count: HashMap<i32, i32>,
```

Comment:
Why HashMap?

Reply:
It was cloned from previous example, what do you think is more appropriate here?

Reply:
I was mislead by `.entry(0)`.